cpmpy/transformations/comparison.py

```diff
@@ -38,18 +38,18 @@ def only_numexpr_equality(constraints, supported=frozenset(), csemap=None):
             if not isinstance(cond, _BoolVarImpl): # expr -> bv
                 res = only_numexpr_equality([cond], supported, csemap=csemap)
                 if len(res) == 1 and cond is not res[0]: # changed, but no new cons due to CSE
```

Comment:
I don't know, this code (and the one before it) seems full of assumptions...

what if len(res) == 1 and cond is res[0]? what happens then, or what should happen then?

is the shortcut 'if len(res) == 1 and res[0] == cond: ...pass/smth`?

and what if len(res)==2, then for some reason you have some strong knowledge on what res[0] will be, and that you only need to have an implication over res[1]? but why?

and also what if len(res) >2?

Reply:
Yeah indeed a lot of assumptions are made here which seem quite unintuitive at first sight.
I think we should refactor this to a separate helper function`_comparison_to_eq` or smt, which can then get called from all places in this "dispatcher" transforamtion.